packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1819,40 +1828,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         if (this.canSendOps()) {
             if (this.context.submitBatchFn !== undefined) {
                 const batchToSend: IBatchMessage[] = [];
+
+                let batchContentsLength = 0;
                 for (const message of batch) {
-                    let contents = message.contents;
-                    let metadata = message.metadata;
-                    if (this.runtimeOptions.compressionOptions.minimumSize &&
-                        this.runtimeOptions.compressionOptions.minimumSize < message.contents.length) {
-                        this.compressedOpCount++;
-                        const copiedMessage = { ...message.deserializedContent };
-
-                        const compressionStart = Date.now();
-                        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(copiedMessage.contents));
-                        const compressedContents = compress(contentsAsBuffer);
-                        const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
-                        const duration = Date.now() - compressionStart;
-
-                        if (this.compressedOpCount % 100) {
-                            this.mc.logger.sendPerformanceEvent({
-                                eventName: "compressedOp",
-                                duration,
-                                sizeBeforeCompression: message.contents.length,
-                                sizeAfterCompression: compressedContent.length,
-                            });
-                        }
-
-                        copiedMessage.contents = compressedContent;
-                        const stringifiedContents = JSON.stringify(copiedMessage);
-
-                        if (stringifiedContents.length < message.contents.length) {
-                            contents = JSON.stringify(copiedMessage);
-                            metadata = { ...message.metadata, compressed: true };
-                        }
-                    }
+                    batchContentsLength += message.contents.length;
+                }
 
-                    batchToSend.push({ contents, metadata });
+                if (this.runtimeOptions.compressionOptions?.minimumSize !== undefined
+                    && this.runtimeOptions.compressionOptions.compressionAlgorithm !== undefined
```

Comment:
It's somewhat weird that we never check what is actual value of compressionAlgorithm here. Hard to say what the future looks like, but if we want to be super generic and extensible here, it should like we should have a factory that gives us OpCompressor (not specific class, but type-erased interface) based on value of compressionAlgorithm

Reply:
I agree. The factory pattern you're suggesting is also what I had in mind here. I can probably get to this today as part of this PR as I address the rest of your comments.